common/logisticspipes/gui/GuiGuideBook.java

```diff
@@ -0,0 +1,331 @@
+		this.drawDefaultBackground();
+		/*GUI Background Layer 0*/
+		GlStateManager.translate(0.0F, 0.0F, 5.0F);
+		GlStateManager.enableBlend();
+		GlStateManager.blendFunc(GL11.GL_SRC_ALPHA, GL11.GL_ONE_MINUS_SRC_ALPHA);
+		mc.renderEngine.bindTexture(GUI_BOOK_TEXTURE);
+		drawScaledCustomSizeModalRect(leftX + borderSize, topY + borderSize, tileSize * 2, 0, tileSize / 2, tileSize / 2, acrossX + tileSize, acrossY + tileSize, textureSide$x, textureSide$y);
+		GlStateManager.popMatrix();
+	}
+
+	@Override
+	public void drawScreen(int mouseX, int mouseY, float partialTicks) {
+		this.drawBackgroundLayer();
+		this.drawTextGui();
+		this.drawForegroundLayer();
+		GlStateManager.pushMatrix();
+		GlStateManager.translate(0.0F, 0.0F, 5.0F);
+		super.drawScreen(mouseX, mouseY, partialTicks);
+		GlStateManager.popMatrix();
+	}
+
+	@Override
+	public void initGui() {
+		// Defining drawing constrains
+		acrossX = this.width * 6 / 8 - 2 * tileSize;
+		acrossY = this.height * 6 / 8 - 2 * tileSize;
+		leftX = this.width * 1 / 8;
+		topY = this.height * 1 / 8;
+		rightX = (leftX + tileSize) + (acrossX); // Does not work
+		bottomY = (topY + tileSize) + (acrossY); // Does work
+		// Setting slider constraints
+		sliderTopY = topY + borderSize;
+		sliderBotY = bottomY + borderSize - sliderHeight;
+		// Setting text constraints
+		maxLength = acrossX + 3 / 2 * borderSize - sliderWidth;
+		textX = leftX + 24;
+		textY = topY + 24;
+		maxLines = (acrossY + borderSize) / 10;
+		// Getting information from item NBT
+		if (hand == EnumHand.MAIN_HAND)
+			book = mc.player.getHeldItemMainhand();
+		else
+			book = mc.player.getHeldItemOffhand();
+		if (book.hasTagCompound()) {
+			NBTTagCompound nbtTagCompound = book.getTagCompound();
+			this.page = nbtTagCompound.getInteger("page");
+			this.sliderProgress = nbtTagCompound.getFloat("sliderProgress");
+		} else {
+			this.page = 1;
+			this.sliderProgress = 0.0F;
+		}
+		// Setting up buttons
+		this.buttonList.clear();
+		this.nextPageBtn = this.addButton(new GuiButton(0, this.width / 2 + 100, this.height - 30, 20, 20, ">"));
+		this.nextPageBtn.enabled = page != pageMax;
+		this.prevPageBtn = this.addButton(new GuiButton(1, this.width / 2 - 120, this.height - 30, 20, 20, "<"));
+		this.prevPageBtn.enabled = page != 1;
+		this.slider = this.addButton(new GuiButton(2, rightX + borderSize - sliderHeight, sliderTopY + (int) ((sliderBotY - sliderTopY) * sliderProgress), sliderWidth, sliderHeight, "|||"));
+		// Splitting current page
+		text = splitLines(StringUtils.translate("book.quickstart." + page), fontRenderer, maxLength);
+	}
+
+	@Override
+	public boolean doesGuiPauseGame() {
+		return false;
+	}
+
+	@Override
+	protected void actionPerformed(GuiButton button) throws IOException {
+		if (button.id == 0) {
+			if (page < pageMax) {
+				page++;
+				prevPageBtn.enabled = true;
+			}
+			if (page == pageMax) nextPageBtn.enabled = false;
+			else nextPageBtn.enabled = true;
```

Comment:
`nextPageBtn.enabled = page != pageMax;`

Reply:
I did it this your way in `initGui()`, I don't even know why I made them like this here...